x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/api/util/installation.ts

```diff
@@ -169,51 +166,60 @@ export const installTranslated = async ({
   securitySolutionContext,
   rulesClient,
   savedObjectsClient,
-  logger,
-}: InstallTranslatedProps) => {
+}: InstallTranslatedProps): Promise<number> => {
   const detectionRulesClient = securitySolutionContext.getDetectionRulesClient();
   const ruleMigrationsClient = securitySolutionContext.getSiemRuleMigrationsClient();
 
-  const { data: rulesToInstall } = await ruleMigrationsClient.data.rules.get(migrationId, {
-    filters: { ids, installable: true },
-    from: 0,
-    size: MAX_TRANSLATED_RULES_TO_INSTALL,
-  });
+  let installedCount = 0;
+  const installationErrors: Error[] = [];
 
-  const { customRulesToInstall, prebuiltRulesToInstall } = rulesToInstall.reduce(
-    (acc, item) => {
-      if (item.elastic_rule?.prebuilt_rule_id) {
-        acc.prebuiltRulesToInstall.push(item);
-      } else {
-        acc.customRulesToInstall.push(item);
-      }
-      return acc;
-    },
-    { customRulesToInstall: [], prebuiltRulesToInstall: [] } as {
-      customRulesToInstall: StoredRuleMigration[];
-      prebuiltRulesToInstall: StoredRuleMigration[];
-    }
-  );
+  // Install rules that matched Elastic prebuilt rules
+  const prebuiltRuleBatches = ruleMigrationsClient.data.rules.searchBatches(migrationId, {
+    filters: { ids, installable: true, prebuilt: true },
+  });
+  let prebuiltRulesToInstall = await prebuiltRuleBatches.next();
+  while (prebuiltRulesToInstall.length) {
+    const { rulesToUpdate, errors } = await installPrebuiltRules(
+      prebuiltRulesToInstall,
+      enabled,
+      securitySolutionContext,
+      rulesClient,
+      savedObjectsClient,
+      detectionRulesClient
+    );
+    installedCount += rulesToUpdate.length;
```

Comment:
@semd fyi I count rules that matched prebuilt rules and those were installed outside the "SIEM Migrations" as successfully installed here. Once we have consistency functionality those rules won't be counted here automatically.

Reply:
that makes sense